src/main/java/uk/gov/hmcts/cmc/claimstore/services/search/DBCaseRepository.java

```diff
@@ -38,13 +39,26 @@ public DBCaseRepository(
     }
 
     @Override
-    public Optional<Claim> linkDefendant(String externalId, String defendantId, String authorisation) {
-        Optional<Claim> claim = claimRepository.getClaimByExternalId(externalId);
-        if (claim.isPresent()) {
-            claimRepository.linkDefendant(claim.orElseThrow(IllegalStateException::new).getId(), defendantId);
-            claim = claimRepository.getClaimByExternalId(externalId);
-        }
+    public Claim linkDefendant(String externalId, String defendantId, String authorisation) {
+        String notFoundErrorMessage = "Claim not found by external id: " + externalId;
+        
+        Claim claim = claimRepository.getClaimByExternalId(externalId)
+            .orElseThrow(() -> new NotFoundException(notFoundErrorMessage));
+        claimRepository.linkDefendant(claim.getId(), defendantId);
+
+        claim = claimRepository.getClaimByExternalId(externalId)
+            .orElseThrow(() -> new NotFoundException(notFoundErrorMessage));;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)